crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -356,9 +356,7 @@ class C:
 
 c_instance = C()
 
-# TODO: Should be `Unknown | int | None`
-# error: [unresolved-attribute]
-reveal_type(c_instance.x)  # revealed: Unknown
+reveal_type(c_instance.x)  # revealed: Unknown | int | None
```

Comment:
We should also have a test similar to this, but with code like `with ContextManager() as (self.x, self.y):`. I think this would catch a change we are missing in the `ast::Expr::Attribute` branch of `SemanticIndexBuilder::visit_expr`, where we match on `Assign` and `For` unpacks, but also need to match on `With`.

Bonus points if we can find a way to make this mistake harder to make in future, e.g. with an `UnpackKind` enum that that code matches on.